torch/_inductor/codegen/scheduling.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+
+class Scheduling(ABC):
```

Comment:
We should modify the existing backends to inherit from this. 

Add docstrings for every method.

Perhaps this belongs in common.py, or some sort of backend registration file.